amundsen_application/dashboard_preview/mode_preview.py

```diff
@@ -40,13 +47,26 @@ def __init__(self, *,
         if has_app_context() and app.config['MODE_REPORT_URL_TEMPLATE'] is not None:
             self._report_url_template = app.config['MODE_REPORT_URL_TEMPLATE']
 
-    @retry(stop_max_attempt_number=3, wait_random_min=500, wait_random_max=1000)
+        self._is_auth_enabled = False
+        if has_app_context() and app.config['ACL_ENABLED_DASHBOARD_PREVIEW']:
+            if not app.config['AUTH_USER_METHOD']:
+                raise Exception('AUTH_USER_METHOD must be configured to enable ACL_ENABLED_DASHBOARD_PREVIEW')
+            self._is_auth_enabled = self.__class__.__name__ in app.config['ACL_ENABLED_DASHBOARD_PREVIEW']
+            self._auth_user_method = app.config['AUTH_USER_METHOD']
+
+    @retry(stop_max_attempt_number=3, wait_random_min=500, wait_random_max=1000,
+           retry_on_exception=_retry_on_retriable_error)
     def get_preview_image(self, *, uri: str) -> bytes:
         """
         Retrieves short lived URL that provides Mode report preview, downloads it and returns it's bytes
         :param uri:
         :return: image bytes
+        :raise: PermissionError when user is not allowed to access the dashboard
         """
+
+        if self._is_auth_enabled:
+            self._authorize_access(user_email=self._auth_user_method(app).email)
```

Comment:
the code lgtm. But it only works for the case when we use email as user id as I believe we allows both email and user_id for primary key for user model.

Reply:
> the code lgtm. But it only works for the case when we use email as user id as I believe we allows both email and user_id for primary key for user model.

Agreed. We probably could refactor this method to add user_id `attribute` later on.

Reply:
So will we have a follow-up on our end to populate `user_id` in our user metadata and use it here instead?

Reply:
For us, it is fine. We populate user_id with https://github.com/lyft/amundsencommon/blob/master/amundsen_common/models/user.py#L59 which will be the same as email. But it won't work for folks if they don't share the same user_id with email.

Reply:
1. Do we have that working? Because when I check my user information my `user_id` is null.
2. If we do have `user_id` then, how come we wouldn't use it right at this moment?

Reply:
@feng-tao Same question as Tamika. Do you know if this is currently working? I assumed that this requires additional effort.

Reply:
Okay, so what would be the list of next steps (for internal & external repos) for being able to switch this to `user_id`? It would be good to have the plan and file a ticket since it sounds like we have to update eventually in order to make this code work for others who request users by explicit `user_id`.

Reply:
@feng-tao @ttannis Confirmed that `user_id` is already working. Posted a PR.
https://github.com/lyft/amundsenfrontendlibrary/pull/485